### PR TITLE
fix(dropdown-filter): disabled can be opened

### PR DESCRIPTION
**Pull request by mJarsater**
**Describe pull-request**  
This PR disables the `<tds-dropdown>` from being opened when disabled. The bug found was that it could be opened by pressing the chevron icon for the `filter` version even if it was `disabled`.

Also added some basic test for regressions on this functionality.

**Solving issue**  
#486 

**How to test**  
1. Go to Dropdown
2. Disable it and set it to filter
3. Try opening it by pressing the icon
4. It should not open.
5. Check out branch
6. Run all tests
7. Make sure they are valid and pass

**Suggested test steps**
- [x] Browser testing (Chrome, Safari, Firefox) 
- [x] Keyboard operability
- [x] Interactive elements have labels.
- [x] Storybook controls
- [x] Design/controls/props is aligned with other components 
- [x] Dark/light mode and variants 
- [x] Input fields – values should be displayed properly 
- [x] Events

**Screenshots**  
_If applicable, add screenshots to help explain_

**Additional context**  
_Add any other context about the pull-request here._

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "start": "stencil build --dev --watch --serve",
     "test": "node docker-test-runner.js",
     "update-snapshots": "docker run --rm --network host -v $(pwd):/work/ -w /work/ mcr.microsoft.com/playwright:v1.39.0-jammy /bin/bash -c 'npm install && npx playwright test --update-snapshots && exit'",
+    "test-mac": "docker run --rm --network host -v $(pwd):/work/ -w /work/ mcr.microsoft.com/playwright:v1.39.0-jammy /bin/bash -c 'npm install && npx playwright test && exit'",
```
Comment (mJarsater) [inline packages/core/package.json:41]:
```suggestion
```

File: packages/core/src/components/accordion/test/expanded/accordion.e2e.ts-snapshots/tds-accordion-renders-expanded-accordion-correctly-1-linux.png
Comment (mJarsater) [inline packages/core/src/components/accordion/test/expanded/accordion.e2e.ts-snapshots/tds-accordion-renders-expanded-accordion-correctly-1-linux.png:1]: Not sure why this is "changed"... 🤔

File: packages/core/src/components/table/table/test/batch/table.e2e.ts-snapshots/tds-table-batch-renders-batch-table-correctly-1-linux.png
Comment (mJarsater) [inline packages/core/src/components/table/table/test/batch/table.e2e.ts-snapshots/tds-table-batch-renders-batch-table-correctly-1-linux.png:1]: Not sure what happened here..

**Comment by mJarsater** [comment]
Marked everyone as assignes, feel free to make the changes you see fit.

Ping @timrombergjakobsson @EmelieLitwin @nathalielindqvist @theJohnnyMe

File: packages/core/src/components/accordion/test/expanded/accordion.e2e.ts-snapshots/tds-accordion-renders-expanded-accordion-correctly-1-linux.png
Comment (mJarsater) [reply packages/core/src/components/accordion/test/expanded/accordion.e2e.ts-snapshots/tds-accordion-renders-expanded-accordion-correctly-1-linux.png:1]: But that change is in the `dropdown-filter.scss`. Can't see why that would impact this screenshot.

Comment (timrombergjakobsson) [reply packages/core/src/components/accordion/test/expanded/accordion.e2e.ts-snapshots/tds-accordion-renders-expanded-accordion-correctly-1-linux.png:1]: Yeah true, visually looks the same.

**Review by theJohnnyMe — APPROVED** [review]
Thanks a lot Martin for contribution in fix and test for it. I did rebase and after it updated 2 screenshots, one for this component and one for link that was off. 
Looks good to me!

**Review by nathalielindqvist — COMMENTED** [review]
One of the tests fail for me, I'm not really sure why?
image

**Review by nathalielindqvist — APPROVED** [review]
Awesome! 🎉

**Pull request merged**